docs/www/data-transform.md

```diff
@@ -0,0 +1,387 @@
+---
+title: Data transformation
+order: 0
+---
+# Data transformation
+
+**Important** - This feature is in technical preview; a cluster with this feature enabled is not yet supported for production.
+
+Redpanda Data Transforms enable users to create WebAssembly-based scripts to perform simple data transformations on topics. 
+
+This eliminates "data ping-pong" by removing the need to consume data streams from a separate system for simple transformations such as scrubbing or cleaning data.
+
+With Redpanda, you can now deploy custom Node.js programs to handle the entire consume-transform-produce cycle for you.
+
+We do all of this co-processing using a new transformation engine built on-top of [WASM (WebAssembly)](https://developer.mozilla.org/en-US/docs/WebAssembly) technology.
+
+All you need to do is to provide:
+- A JavaScript code that transforms the data. 
+- The source topics where to consume from.
+  
+The destination topics can be generated on demand when your transformation runs. 
+
+You can jump start your data transformation development with `rpk`. We provide a sample Node.js project so you can start developing right away. Then you can deploy the data transform to Redpanda for all of the brokers to run.
+
+Be aware that even though we're referring to this feature as data transformation, the Kafka topics are still immutable. 
+What happens under the hood, is that we're reading from an immutable Kafka topic, applying a data transformation process and then we producing to another immutable Kafka topic.
+Your data is never actually transformed, but rather read and written again. 
+
+## Enable data transforms
```

Comment:
@VadimPlh makes a good point that we should suggest that `auto_create_topics_enabled` is set to `False`. It prevents someone from accidentatially creating the expected output topic as a normal topic because your coprocessor attempts to create it from scratch.

If this does occur coproc will complain that you are attempting to push data to a normal topic that already exists. @coral-waters

Reply:
Addressed in my comment on line 33.